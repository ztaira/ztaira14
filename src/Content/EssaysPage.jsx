import React, { Component } from 'react';
import './PageContent.css';
import EssaysItem from './EssaysItem.jsx';

class EssaysPage extends Component {
  constructor(props) {
    super(props);
    this.ReturnEssaysItem = this.ReturnEssaysItem.bind(this);
  }

  ReturnEssaysItem(entry) {
    return <EssaysItem key={entry.name} entry={entry} />;
  }

  render() {
    return (
      <div className="EssaysContent">
        {this.props.allEntries.map(this.ReturnEssaysItem)}
        <div className="RightColumnSpacer" />
      </div>
    );
  }
}

export default EssaysPage;
