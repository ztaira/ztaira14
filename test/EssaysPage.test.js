import React from 'react';
import ReactDOM from 'react-dom';
import { HashRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import EssaysPage from '../src/Content/EssaysPage.jsx';

it('has not changed since the last snapshot', function() {
  const essayspage = renderer.create(
    <HashRouter>
      <EssaysPage allEntries={[
        {
          'name': 'Hello Again, world!',
          'description': 'Testing',
          'file_name': 'hello_again_world.md',
          'create_date': '2017-05-29'
        },
        {
          'name': 'Hello World',
          'description': 'Testing',
          'file_name': 'hello_world.md',
          'create_date': '2017-05-29'
        }
      ]} />
    </HashRouter>
  );
  let essayspage_json = essayspage.toJSON();
  expect(essayspage_json).toMatchSnapshot();
});
