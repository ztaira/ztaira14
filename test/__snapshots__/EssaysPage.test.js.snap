exports[`test has not changed since the last snapshot 1`] = `
<div
  className="EssaysContent">
  <div
    className="EssaysItem">
    <p
      className="EssaysItemDate">
      2017-05-29
    </p>
    <a
      className="EssaysItemLink"
      href="#/Essays/hello_again_world"
      onClick={[Function]}>
      Hello Again, world!
    </a>
    <p
      className="EssaysItemDescription">
      Testing
    </p>
  </div>
  <div
    className="EssaysItem">
    <p
      className="EssaysItemDate">
      2017-05-29
    </p>
    <a
      className="EssaysItemLink"
      href="#/Essays/hello_world"
      onClick={[Function]}>
      Hello World
    </a>
    <p
      className="EssaysItemDescription">
      Testing
    </p>
  </div>
  <div
    className="RightColumnSpacer" />
</div>
`;
